packages/devtools_app/lib/src/connected_app.dart

```diff
@@ -27,8 +29,24 @@ class ConnectedApp {
 
   bool _isFlutterApp;
 
-  Future<bool> get isProfileBuild async =>
-      _isProfileBuild ??= await _connectedToProfileBuild();
+  Future<bool> get isProfileBuild async {
+    // If we have one isolate check if paused on start?
+    final isolates = serviceManager.isolateManager.isolates;
+    if (isolates.length == 1) {
+      final isolate = await serviceManager.service.getIsolate(isolates[0].id);
+      if (isolate.pauseEvent.kind == EventKind.kPauseStart) {
+        // Application started with --start-paused, assume profile build is
```

Comment:
can you file a bug? This seems like a strange assumption. I would suggest we instead fix the _connectedToProfileBuild method. The current implementation is more fragile than it needs to be. Profile builds are missing support like `eval` so we should use that instead of checking for a specific service extension.

Reply:
Filed bug https://github.com/flutter/devtools/issues/2082 and changed to attempt eval if fails assume profile build.